Library/ViewModels/PledgeShippingLocationViewModel.swift

```diff
@@ -89,25 +89,30 @@ public final class PledgeShippingLocationViewModel: PledgeShippingLocationViewMo
     self.shippingLocationButtonTitle = self.notifyDelegateOfSelectedShippingRule
       .map { $0.location.localizedName }
 
-    self.dismissShippingRules = self.dismissShippingRulesButtonTappedProperty.signal
+    self.dismissShippingRules = Signal.merge(
+      self.shippingRulesCancelButtonTappedProperty.signal,
+      self.shippingRuleUpdatedSignal.signal
+        .ignoreValues()
+        .ksr_debounce(.milliseconds(300), on: AppEnvironment.current.scheduler)
```

Comment:
We're using a small delay...using `ksr_debounce` also allows us to test this properly

Reply:
Yes, exactly...we've paired on this with Danny and decided we need a bit of a delay before dismissing so that the user has a perception of what they've selected.

It should still be testable..

```swift
self.vm.inputs.shippingRuleUpdated(to: selectedShippingRule)

self.scheduler.advance(by: .milliseconds(300))

self.dismissShippingRules.assertValueCount(1)
```

Reply:
Awesome. Thanks for clarifying!